core/cli/src/plugin/reduce-installations.ts

```diff
@@ -72,7 +73,10 @@ export async function reducePluginHookInstallations(
         hookConstructor: hookClass
       }
 
-      return hookClass.overrideChildInstallations(plugin, installation, childInstallations)
+      const childInstallationsForHook = childInstallations.filter(
+        (childInstallation) => id === extractForHook(childInstallation)
+      )
```

Comment:
**thought**: this feels like the kind of case where a better data structure would avoid this but 🤷🏻‍♀️